beam-migrate/Database/Beam/Migrate/Log.hs

```diff
@@ -143,7 +145,7 @@ recordCommit commitId = do
 
 -- Ensure the backend tables exist
 ensureBackendTables :: forall be m
-                     . BeamSqlBackendCanSerialize be Text
+                     . (BeamSqlBackendCanSerialize be Text, Fail.MonadFail m)
```

Comment:
Won't this need some CPP to not break with GHC 8.4 and earlier?

Reply:
The three packages I mentioned compile (and test pass, except the ones requiring PostgreSQL server) with GHC-8.0 ... GHC-8.8

This is breaking interface change, but there are no way around it